tests/virtiofs/datavolume.go

```diff
@@ -145,190 +141,191 @@ var _ = Describe("[sig-storage] virtiofs", decorators.SigStorage, func() {
+			if err != nil {
+				return nil
+			}
+			for _, pod := range podList.Items {
+				for _, ownerRef := range pod.GetOwnerReferences() {
+					if ownerRef.UID == vmi.GetUID() {
+						virtlauncherPod = &pod
+						break
 					}
 				}
-				return virtlauncherPod
-			}, 30*time.Second, 1*time.Second).ShouldNot(BeNil())
-			Expect(virtlauncherPod.Spec.Containers).To(HaveLen(4))
-			foundContainer := false
-			virtiofsContainerName := fmt.Sprintf("virtiofs-%s", pvcName)
-			for _, container := range virtlauncherPod.Spec.Containers {
-				if container.Name == virtiofsContainerName {
-					foundContainer = true
-					Expect(container.Resources.Requests.Cpu().Value()).To(Equal(resources.Requests.Cpu().Value()))
-					Expect(container.Resources.Requests.Memory().Value()).To(Equal(resources.Requests.Memory().Value()))
-					Expect(container.Resources.Limits.Cpu().Value()).To(Equal(resources.Limits.Cpu().Value()))
-					Expect(container.Resources.Limits.Memory().Value()).To(Equal(resources.Limits.Memory().Value()))
-				}
-			}
-			Expect(foundContainer).To(BeTrue())
-		})
-	})
-
-	Context("Run a VMI with VirtIO-FS and a datavolume", func() {
-		var dataVolume *cdiv1.DataVolume
-		BeforeEach(func() {
-			checks.SkipTestIfNoFeatureGate(virtconfig.VirtIOFSGate)
-			if !libstorage.HasCDI() {
-				Skip("Skip DataVolume tests when CDI is not present")
 			}
-
-			sc, exists := libstorage.GetRWOFileSystemStorageClass()
-			if !exists {
-				Skip("Skip test when Filesystem storage is not present")
+			return virtlauncherPod
+		}, 30*time.Second, 1*time.Second).ShouldNot(BeNil())
+		Expect(virtlauncherPod.Spec.Containers).To(HaveLen(4))
+		foundContainer := false
+		virtiofsContainerName := fmt.Sprintf("virtiofs-%s", pvcName)
+		for _, container := range virtlauncherPod.Spec.Containers {
+			if container.Name == virtiofsContainerName {
+				foundContainer = true
+				Expect(container.Resources.Requests.Cpu().Value()).To(Equal(resources.Requests.Cpu().Value()))
+				Expect(container.Resources.Requests.Memory().Value()).To(Equal(resources.Requests.Memory().Value()))
+				Expect(container.Resources.Limits.Cpu().Value()).To(Equal(resources.Limits.Cpu().Value()))
+				Expect(container.Resources.Limits.Memory().Value()).To(Equal(resources.Limits.Memory().Value()))
 			}
+		}
+		Expect(foundContainer).To(BeTrue())
+	},
+		Entry("unprivileged virtiofsd", util.NamespaceTestDefault),
+		Entry("privileged virtiofsd", testsuite.NamespacePrivileged),
+	)
+
+	DescribeTable("Run a VMI with VirtIO-FS and a datavolume", func(namespace string) {
+		if namespace == testsuite.NamespacePrivileged {
+			checks.SkipTestIfNoFeatureGate(virtconfig.VirtIOFSGate)
+		}
 
-			dataVolume = libdv.NewDataVolume(
-				libdv.WithRegistryURLSource(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine)),
-				libdv.WithPVC(libdv.PVCWithStorageClass(sc)),
-			)
-		})
+		if !libstorage.HasCDI() {
+			Skip("Skip DataVolume tests when CDI is not present")
```

Comment:
Are the tests expected to run without CDI at all?

Reply:
No, we skip the test if CDI is not present (that check was already in the test before my changes)

Reply:
Yes, my question was more aimed at whether KubeVirt tests run at all without CDI. IOW can we expected CDI always to be present when the tests run (without the need for another check or skip, not present is expected to result in failure)? IIRC it is present by default in kubevirtci.

Reply:
Ohh, I don't know really, maybe this was copy-pasted from `tests/storage/datavolume.go`, that was the original location of this test  (btw, that test checks twice per test if the CDI is present).

I can remove the check if is unnecessary

Reply:
Are these tests only intended for this CI or also for someone who does not have/needs CDI?

Reply:
I can only speak for CNV where CDI is present in downstream tests.